models/user.js

```diff
@@ -24,6 +26,26 @@ const UserSchema = new mongoose.Schema({
     },
 })
 
+UserSchema.pre('save', async function () {
+    const salt = await bcrypt.genSalt(10)
+    this.password = await bcrypt.hash(this.password, salt)
+})
+
+UserSchema.methods.createJWT = function () {
+    return jwt.sign(
+        { userId: this._id, name: this.name },
+        process.env.JWT_SECRET,
```

Comment:
So if you are using these now you could create an `.env.development` file that gets used during local dev, that way we can start using this straight away. You are also able to commit it to github as it would only be test abnd the production one would be `.env` that doesnt get put into git